packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Models/PropertyDescriptionBuilder.cs

```diff
@@ -0,0 +1,184 @@
+                    else
+                    {
+                        description = $"<description>{literalValue}</description>";
+                    }
+                }
+                else
+                {
+                    description = $"<description>{item:C}</description>";
+                }
+
+                values.Add(description);
+            }
+
+            return values.Distinct().ToList();
+        }
+
+        /// <summary>
+        /// This method will construct an additional description for properties that are binary data. For properties whose values are union types,
+        /// the description will include the types of values that are allowed.
+        /// </summary>
+        /// <param name="type">The CSharpType of the property.</param>
+        /// <param name="serializationFormat">The serialization format of the property.</param>
+        /// <returns>The formatted description string for the property.</returns>
+        private static FormattableString CreateBinaryDataExtraDescription(CSharpType type, SerializationFormat serializationFormat)
+        {
+            string typeSpecificDesc;
+            var unionTypes = GetUnionTypes(type);
+            IReadOnlyList<FormattableString> unionTypeDescriptions = Array.Empty<FormattableString>();
+
+            if (unionTypes.Any())
+            {
+                unionTypeDescriptions = GetUnionTypesDescriptions(unionTypes);
+            }
+
+            if (type.FrameworkType == typeof(BinaryData))
+            {
+                typeSpecificDesc = "this property";
+                return ConstructBinaryDataDescription(typeSpecificDesc, serializationFormat, unionTypeDescriptions);
+            }
+            if (type.IsList)
+            {
+                typeSpecificDesc = "the element of this property";
+                return ConstructBinaryDataDescription(typeSpecificDesc, serializationFormat, unionTypeDescriptions);
+            }
+            if (type.IsDictionary)
+            {
+                typeSpecificDesc = "the value of this property";
+                return ConstructBinaryDataDescription(typeSpecificDesc, serializationFormat, unionTypeDescriptions);
+            }
+
+            return FormattableStringHelpers.Empty;
+        }
+
+        // recursively get the union types if any.
+        private static IReadOnlyList<CSharpType> GetUnionTypes(CSharpType type)
+        {
+            if (type.IsCollection)
+            {
+                return GetUnionTypes(type.ElementType);
+            }
+            else if (type.IsUnion)
+            {
+                return type.UnionItemTypes;
+            }
+
+            return Array.Empty<CSharpType>();
+        }
+
+        private static FormattableString ConstructBinaryDataDescription(string typeSpecificDesc, SerializationFormat serializationFormat, IReadOnlyList<FormattableString> unionTypeDescriptions)
+        {
+            FormattableString unionTypesAdditionalDescription = $"";
```

Comment:
maybe we should use `FormattableStringHelpers.Empty` just as what we should do for empty strings.

Reply:
Given the https://github.com/microsoft/typespec/pull/3414#discussion_r1610620834, I will remove this class altogether from SCM library.